python/ray/serve/_private/deployment_state.py

```diff
@@ -1856,3 +1863,8 @@ def update(self):
 
         for tag in deleted_tags:
             del self._deployment_states[tag]
+
+        if len(deleted_tags):
```

Comment:
hmm, do we need to collect the deleted num deployment? this will make the stats inaccurate since you also gather num deployment in the deploy(),

I suggest we don't gather stats in deploy() or delete(), we can have a dedicated stats thread and push the expected stats into the kv store.

(Just checked it is a synchronous call in record_extra_usage_tag. might not be very safe to do it in the critical code path)

Reply:
I'm just trying to capture the "latest" state here since it's last writer win for that hour.